src/glue.c

```diff
@@ -214,6 +217,135 @@ PyObject *mg_path_to_py_path(const mg_path *path) {
+    return (value);                                       \
+  }
+
+PyObject *make_py_date(int y, int m, int d) {
+  PyDateTime_IMPORT;
+  PyObject *date = PyDate_FromDate(y, m, d);
+  if (!date) {
+    PyErr_Print();
+  }
+  return date;
+}
+
+PyObject *make_py_time(int h, int min, int sec, int mi) {
+  PyDateTime_IMPORT;
+  PyObject *time = PyTime_FromTime(h, min, sec, mi);
+  if (!time) {
+    PyErr_Print();
+  }
+  return time;
+}
+
+PyObject *make_py_datetime(int y, int m, int d, int h, int min, int sec,
+                           int mi) {
+  PyDateTime_IMPORT;
+  PyObject *datetime = PyDateTime_FromDateAndTime(y, m, d, h, min, sec, mi);
+  if (!datetime) {
+    PyErr_Print();
+  }
+  return datetime;
+}
+
+PyObject *make_py_delta(int days, int sec, int micros) {
+  PyDateTime_IMPORT;
+  PyObject *delta = PyDelta_FromDSU(days, sec, micros);
+  if (!delta) {
+    PyErr_Print();
+  }
+  return delta;
+}
+
+PyObject *mg_date_to_py_date(const mg_date *date) {
+  PyDateTime_IMPORT;
+  SCOPED_CLEANUP PyObject *unix_epoch = make_py_date(1970, 1, 1);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(unix_epoch, NULL)
+  SCOPED_CLEANUP PyObject *date_as_delta =
+      make_py_delta(mg_date_days(date), 0, 0);
+  SCOPED_CLEANUP PyObject *method_name = PyUnicode_FromString("__add__");
+  PyObject *result_date =
+      PyObject_CallMethodObjArgs(unix_epoch, method_name, date_as_delta, NULL);
+  if (!result_date) {
+    PyErr_Print();
+  }
+  return result_date;
+}
+
+PyObject *mg_local_time_to_py_time(const mg_local_time *lt) {
+  PyDateTime_IMPORT;
+  SCOPED_CLEANUP PyObject *unix_epoch =
+      make_py_datetime(1970, 1, 1, 0, 0, 0, 0);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(unix_epoch, NULL)
+  SCOPED_CLEANUP PyObject *numerator =
+      PyLong_FromLong(mg_local_time_nanoseconds(lt));
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(numerator, NULL)
+  SCOPED_CLEANUP PyObject *denominator = PyLong_FromLong(1000000000);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(denominator, NULL)
+  SCOPED_CLEANUP PyObject *div_result =
+      PyNumber_TrueDivide(numerator, denominator);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(div_result, NULL)
+  SCOPED_CLEANUP PyObject *method_name =
+      PyUnicode_FromString("utcfromtimestamp");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(method_name, NULL)
+  SCOPED_CLEANUP PyObject *result =
+      PyObject_CallMethodObjArgs(unix_epoch, method_name, div_result, NULL);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(result, NULL);
+  SCOPED_CLEANUP PyObject *h = PyObject_GetAttrString(result, "hour");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(h, NULL)
+  SCOPED_CLEANUP PyObject *m = PyObject_GetAttrString(result, "minute");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(m, NULL)
+  SCOPED_CLEANUP PyObject *s = PyObject_GetAttrString(result, "second");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(s, NULL)
+  SCOPED_CLEANUP PyObject *mi = PyObject_GetAttrString(result, "microsecond");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(mi, NULL)
```

Comment:
Why not just divide and mod for all the values? Not sure why use unix_epoch for LocalTime where the epoch is midnight. Also, don't we lose on precision with the floating point numbers?